controllers/userStatus.js

```diff
@@ -70,17 +70,21 @@ const getUserStatus = async (req, res) => {
 const getAllUserStatus = async (req, res) => {
   try {
     const { allUserStatus } = await userStatusModel.getAllUserStatus(req.query);
+    const activeUsers = [];
     for (const status of allUserStatus) {
       //  fetching users from users collection by userID in userStatus collection
       const result = await fetchUser({ userId: status.userId });
-      status.full_name = `${result.user.first_name} ${result.user.last_name}`;
-      status.picture = result.user.picture;
-      status.username = result.user.username;
+      if (!result.user?.roles?.archived) {
+        status.full_name = `${result.user.first_name} ${result.user.last_name}`;
+        status.picture = result.user.picture;
+        status.username = result.user.username;
+        activeUsers.push(status);
+      }
     }
     return res.json({
       message: "All User Status found successfully.",
-      totalUserStatus: allUserStatus.length,
-      allUserStatus,
+      totalUserStatus: activeUsers.length,
+      allUserStatus: activeUsers,
```

Comment:
Good job handling it like this here. This way you saved the extra iterations of changing allUserStatus everywhere it is used

Reply:
Yes changing it in backend was better than handling in frontend everywhere.